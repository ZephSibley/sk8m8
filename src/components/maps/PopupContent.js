import React, { useEffect, useState } from 'react';
import { makeStyles, } from '@material-ui/core/styles';
import CardContent from '@material-ui/core/CardContent';
import CircularProgress from '@material-ui/core/CircularProgress';
import Typography from '@material-ui/core/Typography';
import MarkerStars from './MarkerStars';
// import CardMedia from '@material-ui/core/CardMedia';
// import IconButton from '@material-ui/core/IconButton';
// import SkipPreviousIcon from '@material-ui/icons/SkipPrevious';
// import PlayArrowIcon from '@material-ui/icons/PlayArrow';
// import SkipNextIcon from '@material-ui/icons/SkipNext';



const PopupContent = props => {
    // prop: markerId; int
    // prop: requests; http client

    const useStyles = makeStyles(theme => ({
        card: {
            display: 'flex',
            minWidth: 301
        },
        details: {
            display: 'flex',
            flexDirection: 'column',
        },
        content: {
            flex: '1 0 auto',
        },
        cover: {
            width: 151,
        },
        controls: {
            display: 'flex',
            alignItems: 'center',
            paddingLeft: theme.spacing(1),
            paddingBottom: theme.spacing(1),
        },
        playIcon: {
            height: 38,
            width: 38,
        },
    }));

    const classes = useStyles();

    const [markerDetails, updateMarkerDetails] = useState({
        name: <CircularProgress />,
        locationCategory: '',
        username: '',
        starCount: 0,
        hasStarred: false,
    })

    useEffect(() => {
        props.requests.get(
            `${process.env.REACT_APP_ENDPOINT}mapmarker/details?id=${props.markerId}`,
            {
                headers: {
                    'Authorization': `Bearer ${localStorage.getItem('token')}`
                }
            }
        ).then(response => 
            updateMarkerDetails(response.data)
        ).catch(e => {
            e.response && e.response.status === 401 ?
                updateMarkerDetails({
                    name: '',
                    locationCategory: <div>Please <a href='/#/login'>Log in</a> or <a href='/#/signup'>sign up</a> to view marker details</div>,
                    username: '',
                    starCount: 0,
                    hasStarred: false,
                })
                :
                updateMarkerDetails(m => m.name = e.message)
        });
    }, [props.requests, props.markerId]);

    return (
        <div className={classes.card}>
            <div className={classes.details}>
                <CardContent className={classes.content}>
                    <Typography component="h5" variant="h5">
                        {markerDetails.name}
                    </Typography>
                    <Typography variant="subtitle1" color="textSecondary">
                        {markerDetails.locationCategory}
                    </Typography>
                    <Typography variant="subtitle2" color="textSecondary">
                        {markerDetails.username ? `Created by ${markerDetails.username}` : ''}
                    </Typography>
                    <MarkerStars 
                        requests={props.requests}
                        markerId={props.markerId}
                        starCount={markerDetails.starCount}
                        hasStarred={markerDetails.hasStarred}
                    />
                </CardContent>
                {/*
                    <div className={classes.controls}>
                        <IconButton aria-label="previous">
                            {theme.direction === 'rtl' ? <SkipNextIcon /> : <SkipPreviousIcon />}
                        </IconButton>
                        <IconButton aria-label="play/pause">
                            <PlayArrowIcon className={classes.playIcon} />
                        </IconButton>
                        <IconButton aria-label="next">
                            {theme.direction === 'rtl' ? <SkipPreviousIcon /> : <SkipNextIcon />}
                        </IconButton>
                    </div>
                */}
            </div>
            {/*
                <CardMedia
                    className={classes.cover}
                    image={CircularProgress}
                    title="video"
                />
            */}
        </div>
    );
}

export default PopupContent;