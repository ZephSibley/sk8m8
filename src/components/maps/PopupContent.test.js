import React from 'react';
import renderer from 'react-test-renderer';
import mockAxios from '../../utils/__mocks__/mockAxios';
import PopupContent from './PopupContent';

test('Basic Rendering', () => {
    mockAxios.get.mockImplementationOnce(() =>
        Promise.resolve({
            data: {
                name: 'Name!',
                locationType: 'Skatey',
                creator: 'Zeph!'
            }
        })
    )

    const component = renderer.create(
        <PopupContent 
            markerId={1}
            requests={mockAxios}
        />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})