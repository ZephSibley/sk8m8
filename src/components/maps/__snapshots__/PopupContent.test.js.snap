// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
<div
  className="makeStyles-card-1"
>
  <div
    className="makeStyles-details-2"
  >
    <div
      className="MuiCardContent-root makeStyles-content-3"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5"
      >
        Loading
      </h5>
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
      >
        
      </h6>
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
      >
        Created by [object Object]
      </h6>
    </div>
    <div
      className="makeStyles-controls-5"
    >
      <button
        aria-label="previous"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 6h2v12H6zm3.5 6l8.5 6V6z"
            />
          </svg>
        </span>
      </button>
      <button
        aria-label="play/pause"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-playIcon-6"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M8 5v14l11-7z"
            />
          </svg>
        </span>
      </button>
      <button
        aria-label="next"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 18l8.5-6L6 6v12zM16 6v12h2V6h-2z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiCardMedia-root makeStyles-cover-4"
    style={
      Object {
        "backgroundImage": "url(\\"ajax-loader.gif\\")",
      }
    }
    title="video"
  />
</div>
`;
